include/envoy/ratelimit/ratelimit.h

```diff
@@ -34,5 +38,48 @@ struct Descriptor {
   absl::optional<RateLimitOverride> limit_ = absl::nullopt;
 };
 
+/**
+ * A single token bucket. See token_bucket.proto.
+ */
+struct TokenBucket {
+  uint32_t max_tokens_;
+  uint32_t tokens_per_fill_;
+  std::chrono::milliseconds fill_interval_;
+
+  friend bool operator==(const TokenBucket& lhs, const TokenBucket& rhs) {
+    return lhs.max_tokens_ == rhs.max_tokens_ && lhs.tokens_per_fill_ == rhs.tokens_per_fill_ &&
+           lhs.fill_interval_ == rhs.fill_interval_;
+  }
+
+  // Support absl::Hash.
+  template <typename H>
+  friend H AbslHashValue(H h, const TokenBucket& d) { // NOLINT(readability-identifier-naming)
+    h = H::combine(std::move(h), d.max_tokens_, d.tokens_per_fill_);
```

Comment:
Missing interval

Reply:
Didn't add before as absl::Hash is not available for std::chrono, but I have now changed this to absl::Duration